test/goal_test.exs

```diff
@@ -596,6 +598,24 @@ defmodule GoalTest do
       assert errors_on(changeset_2) == %{id: ["has invalid format"]}
     end
 
+    test "string, format: custom regex" do
+      Application.put_env(:goal, :custom_regex, ~r/^[a-fA-F0-9]{40}$/)
+
+      schema = %{hash: [type: :string, format: :custom]}
+
+      hash =  :crypto.hash(:sha, "test") |> Base.encode16
+      data_1 = %{"hash" => hash}
+      data_2 = %{"hash" => "notahash"}
+
+      changeset_1 = Goal.build_changeset(schema, data_1)
+      changeset_2 = Goal.build_changeset(schema, data_2)
+
+      assert changes_on(changeset_1) == %{hash: hash}
+      assert errors_on(changeset_2) == %{hash: ["has invalid format"]}
+
+      Application.delete_env(:goal, :custom_regex)
+    end
```

Comment:
I've choose to do a hash regex validation here, not sure if this is the best approach, lmk